recipe/recipe-server.go

```diff
@@ -0,0 +1,100 @@
+package recipe
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/peppermint-recipes/peppermint-server/database"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var (
+	errRecipeIsNotValid = errors.New("recipe is not valid")
+)
+
+type recipeServer struct {
+	mongoClient *mongo.Client
+}
+
+func NewRecipeServer() *recipeServer {
+	mongoClient, _, _ := database.GetConnection()
+	return &recipeServer{mongoClient: mongoClient}
+}
+
+func (rs *recipeServer) GetAllRecipesHandler(context *gin.Context) {
+	var loadedTasks, err = getAllRecipes()
+	if err != nil {
+		context.JSON(http.StatusNotFound, gin.H{"message": err})
+		return
+	}
+	context.JSON(http.StatusOK, gin.H{"recipes": loadedTasks})
+}
+
+func (rs *recipeServer) GetRecipeByIDHandler(context *gin.Context) {
+	recipeID := context.Param("id")
+
+	var loadedRecipe, err = getRecipeByID(recipeID)
+	if err != nil {
+		context.JSON(http.StatusNotFound, gin.H{"message": err})
+
+		return
+	}
+	context.JSON(http.StatusOK, gin.H{"Recipe": loadedRecipe})
+}
+
+func (rs *recipeServer) CreateRecipeHandler(context *gin.Context) {
+	var recipe Recipe
+
+	if err := context.ShouldBindJSON(&recipe); err != nil {
+		context.String(http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	if !recipe.isValid() {
+		context.String(http.StatusBadRequest, errRecipeIsNotValid.Error())
+
+		return
+	}
+
+	recipe.LastUpdated = time.Now()
+
+	id, err := createRecipe(&recipe)
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{"message": err})
+		return
+	}
+	context.JSON(http.StatusOK, gin.H{"id": id})
```

Comment:
Since we are updating recipes lastUpdated, we should return the whole recipe to avoid unnecessary diffs between update times betwen server and client.